### DISCUSSION TITLE
Wayland/wlroots bindings

### DISCUSSION THREAD

**Discussion by smhcpp** [vlang/v#15158] [General]
I am trying to write a wayland compositor based on wlroots. I am trying to use v-translate to write wayland, wlroots, pixman v bindings. I get errors all the time. Have anybody tried writing these bindings so far? Any hints to how I could follow up?

**Comment by medvednikov** [comment]
Please report these errors via github.com/vlang/c2v

They will be fixed.

**Reply by smhcpp** [reply]
Thanks for the reply. Just to be sure it is an error, I cloned the wayland library and ran `v translate wayland` it gave an error _No input file_.
How can I report the error exactly? because I tried with `v -d trace_verbose` and it responded _Unknown command_.

**Reply by medvednikov** [reply]
You need to provide the directory that has .c files. Maybe it's it's in `wayland/src/` ?

**Reply by smhcpp** [reply]
I tried `v translate wayland/src` and it gave me:
_translating_ `./wayland-shm.c ... ./wayland-shm.c:33:10:` _fatal error:_ `'config.h'` _file not found_
`#include "config.h"`
      `^~~~~~~~~~`
_1 error generated._
_The file_ `./wayland-shm.c` _could not be parsed as a C source file._

**Reply by larpon** [reply]
Missing config.h files is often because you need to run a configure script that generates the config.h for your platform. If you can't build the project, c2v can't translate it.

**Reply by smhcpp** [reply]
Thanks. I had some issues with building wayland with meson. I think I got it right as it says _Directory already configured_. Then I copied the config file `wayland/build/config.h` to `wayland/src` and now there is no error of `config.h` _file not found_. However, there is a new issue  now starting with `Unhandled expr() node {ReturnStmt}`. Should I put it here ( around 20 lines of error )?
